GVRf/Framework/jni/objects/light.h

```diff
@@ -92,6 +92,155 @@ class Light: public HybridObject {
+        spotangle = angle;
+    }
+
+    float getSpotangle() const {
+        return spotangle;
+    }
+
+    void setOverwriteTextureRender(bool enable) {
+        overwriteTextureRender = enable;
+    }
+
+    float isOverwriteTextureRender() const {
+        return overwriteTextureRender;
+    }
+
+    int getLightType() const {
+        return type;
+    }
+
+    void setLightType(int light_type) {
+        type = light_type;
+    }
+
+    void setShadowMapHandlerMode(ShadowMapHandlerMode mode) {
+        shadowmap_handler = mode;
+    }
+
+    ShadowMapHandlerMode getShadowMapHandlerMode() const {
+        return shadowmap_handler;
+    }
+
+    float getShadowSmoothSize() const {
+        return shadow_smooth_size;
+    }
+
+    void setShadowSmoothSize(float size) {
+        shadow_smooth_size = size;
+    }
+
+    void setShadowMapHandlerEdges(ShadowMapHandlerEdges mode) {
+        shadowmap_handler_edges = mode;
+    }
+
+    ShadowMapHandlerEdges getShadowMapHandlerEdges() const {
+        return shadowmap_handler_edges;
+    }
+
+    float getShadowMapEdgesLength() const {
+        return shadowmap_edges_length;
+    }
+
+    void setShadowMapEdgesLength(float value) {
+        shadowmap_edges_length = value;
+    }
+
+    float getLightAmbientOnShadow() const {
+        return light_ambient_on_shadow;
+    }
+
+    void setLightAmbientOnShadow(float value) {
+        light_ambient_on_shadow = value;
+    }
+
+    float getShadowGradientCenter() const {
+        return shadow_gradient_center;
+    }
+
+    void setShadowGradientCenter(float value) {
+        shadow_gradient_center = value;
+    }
+
+    float getLightingShade() const {
+        return u_lighting_shade;
+    }
+
+    void setLightingShade(float value) {
+        u_lighting_shade = value;
+    }
+
+    float getBoardStratifiedSampling() const {
+        return u_stratified_sampling;
+    }
+
+    void setBoardStratifiedSampling(float value) {
+        u_stratified_sampling = value;
+    }
+
+    float getShadowSmoothDistance() const {
+        return u_shadow_smooth_distance;
+    }
+
+    void setShadowSmoothDistance(float value) {
+        u_shadow_smooth_distance = value;
+    }
+
+    float getBias() const {
+        return bias;
+    }
+
+    void setBias(float value) {
+        bias = value;
+    }
+
+private:
+    glm::vec3 light_position_;
+    glm::vec3 light_direction_ = glm::vec3(0);
+
+    int type = POINT;
+
+    bool overwriteTextureRender = false;
+    float spotangle = 100;
+    ShadowMapHandlerMode shadowmap_handler = HIDE;
+    ShadowMapHandlerEdges shadowmap_handler_edges = HANDLER_EDGES_DARK;
+    float shadow_smooth_size = 0.0;
+    float shadowmap_edges_length = 0.1;
+    float light_ambient_on_shadow = 0.1;
+    float shadow_gradient_center = 2.0;
+    float u_lighting_shade = 0.0;
+    float u_stratified_sampling = 1.0;
+    float u_shadow_smooth_distance = 1.0;
+    float bias = 0.001;
+    // ========== LIGHT parameters END ==========
+
```

Comment:
Some design questions :- 

1. This design mixes two modes -- we have a map to store arbitrary parameters and we have explicitly named parameters. (eg) Position is stored in the map while direction is a named parameter. What controls this decision? As an example, if we decide to define only common parameters (that is used in all light types) as named parameters, then all the shadow specific parameters should be placed in the map. 
2. A higher level design issue is how we plan to use this class? We will have a switch stmt (switch (cameraLight->getLightType()) ) and each case would either get parameters from the map or the named parameters for that light type. As a app developer, I would not know what parameters are applicable for a particular light unless I look at comments in the Light.h file (assuming it is kept up to date with changes). Would it be preferable to derive the different Light types from the Light class so it is clear in the class definition what parameters are applicable for that Light or shadow or ...?